types/net_trigger.d.ts

```diff
@@ -1,12 +1,12 @@
 import { NetFields } from './net_fields';
 import { NetMatches } from './net_matches';
 
-export class CactbotRegExpExecArray<T extends TriggerTypes> extends RegExpExecArray {
+export type CactbotRegExpExecArray<T extends TriggerTypes> = RegExpExecArray & {
   groups?: NetMatches[T];
-}
+};
 
-export class CactbotBaseRegExp<T extends TriggerTypes> extends RegExp {
-  exec(string: string): CactbotRegExpExecArray<T> | null;
-}
+export type CactbotBaseRegExp<T extends TriggerTypes> = RegExp & {
+  exec: (string: string) => CactbotRegExpExecArray<T> | null;
+};
```

Comment:
It is weird that here define classes and `extends` an interface (theoretically an interface should only be `implement`-ed when you using it in a class)
so I change to a `type` expression and do a union operation.